Extractor/ServerSubscriptionManager.cs

```diff
@@ -0,0 +1,106 @@
+        private readonly UAClient _uaClient;
+        private readonly ServerNamespacesToRebrowseConfig? _config;
+
+        private readonly uint[] attributes = new[]
+        {
+            Attributes.NodeId,
+            Attributes.DisplayName,
+            Attributes.DataType,
+            Attributes.NodeClass,
+            Attributes.Description
+        };
+
+        public ServerSubscriptionManager(
+            ILogger<ServerSubscriptionManager> logger,
+            UAClient uaClient,
+            ServerNamespacesToRebrowseConfig? config
+        )
+        {
+            _logger = logger;
+            _uaClient = uaClient;
+            _config = config;
+        }
+
+        public async Task EnableCustomServerSubscriptions(UAExtractor extractor, CancellationToken token)
+        {
+            var serverNode = await _uaClient.GetServerNode(token);
+
+            List<NodeId> nodeIds = new List<NodeId>();
+            var targets = _config?.Namespaces;
+            // nodeIds.Add("ns=1;s=Input4");
+
+            await _uaClient.Browser.BrowseDirectory(
+                new[] { serverNode.Id },
+                (refDef, id) =>
+                {
+                    var nodeId = (NodeId)refDef.NodeId;
+                    if (targets.Contains(refDef.DisplayName.ToString()))
+                    {
+                        nodeIds.Add(nodeId);
+                        _logger.LogInformation(
+                            "Subscription to a rebrowse on node {node} with {id} is now set",
+                            refDef.DisplayName.ToString(), nodeId
+                        );
+                    }
+                },
+                token,
+                maxDepth: -1,
+                doFilter: false
+            );
+
+            var readValueIds = new ReadValueIdCollection(
+                nodeIds.SelectMany(
+                    node => attributes.Select(att => new ReadValueId { AttributeId = att, NodeId = node })
+                )
+            );
+            var results = await _uaClient.ReadAttributes(readValueIds, nodeIds.Count, token, "server subscriptions");
+            var nodes = new Dictionary<NodeId, UAHistoryExtractionState>();
+
+            for (var id = 0; id < nodeIds.Count; id++)
+            {
+                var nc = (NodeClass)results[id * attributes.Count() + 3].Value;
+                if (nc != NodeClass.Variable) continue;
+                var rawDt = results[id * attributes.Count() + 2].GetValue(NodeId.Null);
+
+                nodes[nodeIds[id]] = new ServerItemSubscriptionState(_uaClient, nodeIds[id]);
+            }
+
+            await _uaClient.AddSubscriptions(nodes.Values, "NodeMetrics",
+               (MonitoredItem item, MonitoredItemNotificationEventArgs _) => extractor.Looper.QueueRebrowse(),
+                state => new MonitoredItem
+                {
+                    StartNodeId = state.SourceId,
+                    SamplingInterval = 10,
```

Comment:
This is even lower, isn't it? I think this is in milliseconds, so now it's 100 times a second, might be a bit much.